### DISCUSSION TITLE
How to update the default placeholder and label of a trait via the TraitManager

### DISCUSSION THREAD

**Discussion by Metaphysics0** [GrapesJS/grapesjs#4773] [Q&A]
I am simply trying to update the label and placeholder of the `Href` trait, however when I extend the link it causes the default behavior to break.

![Screen Shot 2022-12-08 at 14 02 13](https://user-images.githubusercontent.com/47161696/206442017-820b3181-fbac-4d42-a0e4-3b8cdf0f6825.png)

Let me know what the proper way of doing this will be

What I tried:
```
const domc = editor.DomComponents

domc.addType("link", {
    isComponent: (el: any): boolean => (el.tagName || "").toLowerCase() == "a",
    extend: "link",
    model: {
      defaults: {
        droppable: true,
      },
    placeholder: "..."
    },
  });
```
Doesn't change anything 👆

```
const domc = editor.DomComponents

domc.addType("link", {
    isComponent: (el: any): boolean => (el.tagName || "").toLowerCase() == "a",
    extend: "link",
    model: {
      defaults: {
        droppable: true,
        traits: [
          {
            type: "text",
            label: "Link (partial or full URL)",
            placeholder: "e.g. /shop/?type=wine or https://google.com",
          },
        ],
      },
    },
  });
```
This overrides the actual behavior of the link.👆

Thanks so much.

**Comment by artf** [comment]
[Here](https://grapesjs.com/docs/modules/Traits.html#add-traits-to-components) you can see how to define default values (value is taken from the component itself)
